sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -618,19 +618,36 @@ object FunctionRegistry {
           }
           throw new AnalysisException(invalidArgumentsMsg)
         }
-        Try(f.newInstance(expressions : _*).asInstanceOf[Expression]) match {
-          case Success(e) => e
-          case Failure(e) =>
-            // the exception is an invocation exception. To get a meaningful message, we need the
-            // cause.
-            throw new AnalysisException(e.getCause.getMessage)
+        try {
+          f.newInstance(expressions : _*).asInstanceOf[Expression]
+        } catch {
+          // the exception is an invocation exception. To get a meaningful message, we need the
+          // cause.
+          case e: Exception => throw new AnalysisException(e.getCause.getMessage)
         }
       }
     }
 
     (name, (expressionInfo[T](name), builder))
   }
 
+  private def expressionWithAlias[T <: Expression](name: String)
+      (implicit tag: ClassTag[T]): (String, (ExpressionInfo, FunctionBuilder)) = {
+    val constructors = tag.runtimeClass.getConstructors
+      .filter(_.getParameterTypes.head == classOf[String])
+    assert(constructors.length == 1)
+    val builder = (expressions: Seq[Expression]) => {
+      try {
+        constructors.head.newInstance(name.toString, expressions.head).asInstanceOf[Expression]
```

Comment:
Since, we are not validating arguments, queries like
`SELECT EVERY(true, false);` will result `true`.

Reply:
We can validate arguments with assert or as used in `expression`?
cc @cloud-fan @HyukjinKwon

Reply:
how is it done in `def expression`?

Reply:
https://github.com/apache/spark/blob/ebd83a544e0eb9fe03e9c1c879e00b50d947a761/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala#L602-L620

Reply:
Since, in `expressionWithAlias` we are always passing `expressions.head` to function's constructor. We can use assert statement
```
...
val builder = (expressions: Seq[Expression]) => {
      assert(expressions.size == 1,
        s"Invalid number of arguments for function $name. " +
          s"Expected: 1; Found: ${expressions.size}")
      assert(expressions.head == classOf[Expression],
        s"Invalid arguments for function $name")
      try {
        constructors.head.newInstance(name.toString, expressions.head).asInstanceOf[Expression]
      }
...
```

Reply:
cc @cloud-fan

Reply:
SGTM

Reply:
BTW is it possible to do `newInstance(name.toString, expressions: _*)`? Then it can work for other expressions that take more than 1 parameter.

Reply:
> is it possible to do newInstance(name.toString, expressions: _*)?

No, compilation error.

Reply:
how about `newInstance((name.toString +: expressions): _*)`?

Reply:
It works